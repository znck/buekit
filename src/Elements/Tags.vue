<script>
import { style, styleResolver } from '../utils'
import PropTypes from '@znck/prop-types'

export default {
  name: 'Tags',
  functional: true,
  props: {
    tag: PropTypes.string.value('span'),
    addons: PropTypes.bool.value(false)
  },
  render (h, ctx = {}) {
    const s = styleResolver(ctx.$style)
    const { addons, tag } = ctx.props

    const styles = [
      s('tags'),
      addons && s('has-addons')
    ]

    return style(h(tag, ctx.data, ctx.children), styles)
  }
}
</script>

<style lang="scss" module>
@import '~bulma/sass/utilities/_all.sass';
@import '../_base.scss';
@import '~bulma/sass/elements/tag.sass';
</style>
