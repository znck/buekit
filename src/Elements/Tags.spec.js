import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'

import Tags from './Tags.vue'

describe('Tags.vue', () => {

  it ('matches snapshot', () => {
    const wrapper = shallow(Tags, {})
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
