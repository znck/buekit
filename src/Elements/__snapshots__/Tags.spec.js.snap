// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags.vue matches snapshot 1`] = `<span class="tags"></span>`;
